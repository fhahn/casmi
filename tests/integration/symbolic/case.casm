function (symbolic) a: -> Int

rule main =
{
  
  case a of
    1: a := 1
    2: a := 2
    default: a := 3
  endcase
  program(self) := undef
}

init main

