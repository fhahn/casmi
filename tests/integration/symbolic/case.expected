forklog:0
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,sta(1,sym2)).%CREATE: a
fof('idcase.casm:7',hypothesis,sym2=1).
fof(id1,hypothesis,sta(2,1)).%UPDATE: a
fof(final0,hypothesis,sta(0,1)).%FINAL: a

forklog:1
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,sta(1,sym2)).%CREATE: a
fof('idcase.casm:8',hypothesis,sym2=2).
fof(id1,hypothesis,sta(2,2)).%UPDATE: a
fof(final0,hypothesis,sta(0,2)).%FINAL: a

forklog:D
tff(symbolNext, type, sym2: $int).
fof(id0,hypothesis,sta(1,sym2)).%CREATE: a
fof(id1,hypothesis,sta(2,3)).%UPDATE: a
fof(final0,hypothesis,sta(0,3)).%FINAL: a

